packages/flutter/lib/src/painting/borders.dart

```diff
@@ -422,17 +468,22 @@ class Border {
     BorderRadius borderRadius,
   }) {
     if (isUniform) {
-      if (shape == BoxShape.circle) {
-        assert(borderRadius == null, 'A borderRadius can only be given for rectangular boxes.');
-        _paintUniformBorderWithCircle(canvas, rect);
-        return;
+      switch (top.style) {
+        case BorderStyle.none:
```

Comment:
Can we have `top.style` be `none` but have a solid style on another edge?

Reply:
not if isUniform is true